allen-cell-animated/nucmorph-colorizer
# fix: Fixed assets not loading on GH pages or internal builds

Problem
=======
Changes almost all image assets to be loaded via `import` syntax, instead of using. This allows vite to transform them properly!

This fixes #277, where font and video files were not being loaded on GitHub pages.

*Estimated review size: small, 5 minutes. (Most of the changes are just file moves!)*

Solution
========
- Switches assets to use `vite`'s bundler/asset management
- Moves the font and video files to the `assets` folder, and updates their imports.
- Fixes a few places in the code where SVG assets were being loaded via URL.
- Adds lazy-loading for the video so it does not block initial render.

## Type of change
* Bug fix (non-breaking change which fixes an issue)

Steps to Verify:
----------------
1. Open the PR preview here, then go to the console: https://allen-cell-animated.github.io/nucmorph-colorizer/pr-preview/pr-314/
2. You should not see any load-related error messages!
3. Also, the video on the landing page should play correctly.

Screenshots (optional):
-----------------------
![image](https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/00092b78-8ef5-42d9-9f11-a258bb2aea51)
Video banner is animating!
